gen.go

```diff
@@ -938,7 +946,38 @@ func emitCborUnmarshalStructField(w io.Writer, f Field) error {
 
 func emitCborUnmarshalInt64Field(w io.Writer, f Field) error {
 	return doTemplate(w, f, `{
-	maj, extra, err := {{ ReadHeader "cr" }}
+	{{ if .Pointer }}
+	b, err := cr.ReadByte()
+	if err != nil {
+		return err
+	}
+	if b != cbg.CborNull[0] {
+		if err := cr.UnreadByte(); err != nil {
+			return err
+		}
+		maj, extra, err = {{ ReadHeader "cr" }}
+		var extraI int64
+		if err != nil {
+			return err
+		}
+		switch maj {
```

Comment:
Can we avoid duplicating this logic? I.e., reuse the same code below and have a second "if pointer" check?

Reply:
👍